test/metric_value_benchmark/collectd_test.go

```diff
@@ -0,0 +1,98 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+//go:build linux && integration
+// +build linux,integration
+
+package metric_value_benchmark
+
+import (
+	"time"
+
+	"github.com/aws/amazon-cloudwatch-agent-test/internal/common"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/metric"
+	"github.com/aws/amazon-cloudwatch-agent-test/test/status"
+)
+
+type CollectDTestRunner struct {
+	BaseTestRunner
+}
+
+var _ ITestRunner = (*CollectDTestRunner)(nil)
+
+func (t *CollectDTestRunner) validate() status.TestGroupResult {
+	metricsToFetch := t.getMeasuredMetrics()
+	testResults := make([]status.TestResult, len(metricsToFetch))
+	for i, metricName := range metricsToFetch {
+		testResults[i] = t.validateCollectDMetric(metricName)
+	}
+
+	return status.TestGroupResult{
+		Name:        t.getTestName(),
+		TestResults: testResults,
+	}
+}
+
+func (t *CollectDTestRunner) getTestName() string {
+	return "CollectD"
+}
+
+func (t *CollectDTestRunner) getAgentConfigFileName() string {
+	return "collectd_config.json"
+}
+
+func (t *CollectDTestRunner) getAgentRunDuration() time.Duration {
+	return minimumAgentRuntime
+}
+
+func (t *CollectDTestRunner) setupAfterAgentRun() error {
+	// EC2 Image Builder creates the collectd's default configuration and collectd will pick it up.
+	// For Linux the static is at /etc/collectd.conf, fox Ubuntu it is at /etc/collectd/collectd.conf
+	// Collectd's static configuration
+	//		LoadPlugin network
+	//		LoadPlugin cpu
+	// 		<Plugin cpu>
+	//			ReportByState = true
+	//			ReportByCpu = true
+	//			ValuesPercentage = true
+	//		</Plugin>
+	//		<Plugin network>
+	//			Server "127.0.0.1" "25826"
+	//		</Plugin>
+	startCollectdCommands := []string{
+		"sudo systemctl restart collectd",
```

Comment:
Why do we need to restart collectd?

Reply:
Related, do we care about stopping it towards the end of the test run?

Reply:
At the end, agent will receive new config. Therefore, it does not matter for me IMO. No particular reason, i guess I am only want to make sure collectd is starting after all. And if its fails, its will return some error during running github action.